### PR TITLE
update: rewriting libraries for data analysis

### PR DESCRIPTION
**Pull request by AlejandraPedroza**
Rewriting the content and formatting it into a table

### CODE REVIEW

File: docs/topics/data-analysis/data-science-libraries.md [hunk 1]
```diff
@@ -1,88 +1,43 @@
...
+| [londogard-nlp-toolkit](https://github.com/londogard/londogard-nlp-toolkit/) | - Data cleaning and processing <br/> - Model building                                                                    | - Utilities for natural language processing <br/> - Library written in Kotlin for the JVM <br/> - Word embeddings, stopwords, stemming, tokenizers, vectorizers, and more processing tools <br/> - Ready-made algorithms for machine learning and deep learning                                                                                                                                                                                                                                                                                                                                                                                                  |
```
Comment (AlejandraPedroza) [inline docs/topics/data-analysis/data-science-libraries.md:17]: Note: The last code update for this library was in 2022

File: docs/topics/data-analysis/data-science-libraries.md [hunk 2]
```diff
@@ -1,88 +1,43 @@
...
+| [Tablesaw](https://github.com/jtablesaw/tablesaw)                      | - Data collection <br/> - Data cleaning and processing <br/> - Data exploration and visualization | - Tools for loading, cleaning, transforming, filtering, and summarizing data <br/> -  Inspired by [Plot.ly](https://plotly.com/)                                                                                                                                                                                                                                                                            |
+| [Dex](https://github.com/PatMartin/Dex)                                | - Data collection <br/> - Data cleaning and processing <br/> - Data exploration and visualization | - Tools for reading, transforming, and visualizing data <br/> - Written in Groovy and Java on top of [JavaFX](https://openjfx.io/)                                                                                                                                                                                                                                                                          |
```
Comment (AlejandraPedroza) [inline docs/topics/data-analysis/data-science-libraries.md:31]: Note: The last update for this library was 5 years ago

File: docs/topics/data-analysis/data-science-libraries.md [hunk 3]
```diff
@@ -1,88 +1,43 @@
...
+| [Smile](https://github.com/haifengl/smile)                             | - Data cleaning and processing <br/> - Data exploration and visualization <br/> - Model building  | - Ready-made algorithms for machine learning and natural language processing <br/> - Linear algebra, graph, interpolation, and visualization tools <br/> - Smile provides a functional [Kotlin API](https://github.com/haifengl/smile/tree/master/kotlin), [Scala API](https://github.com/haifengl/smile/tree/master/scala), [Clojure API](https://github.com/haifengl/smile/tree/master/clojure), and more |
+| [Smile-NLP-kt](https://github.com/londogard/smile-nlp-kt)              | Data cleaning and processing                                                                      | - Kotlin rewrite of the [Scala](https://www.scala-lang.org/api/current/) implicits for the natural language processing part of Smile <br/> - Operations in the format of Kotlin extension functions and interfaces <br/> - Sentence breaking, stemming, bag of words, and other tasks                                                                                                                       |
```
Comment (AlejandraPedroza) [inline docs/topics/data-analysis/data-science-libraries.md:34]: Note: The last code update for this library was 4 years ago

File: docs/topics/data-analysis/data-science-libraries.md [hunk 4]
```diff
@@ -1,88 +1,43 @@
...
+| [Weka](https://www.cs.waikato.ac.nz/ml/index.html)                     | - Data cleaning and processing <br/> - Data exploration and visualization  <br/> - Model building | - Collection of machine learning algorithms for data mining tasks   <br/> - Tools for data preparation, classification, regression, clustering, association rules mining, and visualization                                                                                                                                                                                                                 |
```
Comment (AlejandraPedroza) [inline docs/topics/data-analysis/data-science-libraries.md:35]: Note: The last version of this library was released in 2021

File: docs/topics/data-analysis/data-science-libraries.md [hunk 5]
```diff
@@ -1,88 +1,43 @@
...
+| [Apache OpenNLP](https://opennlp.apache.org/)                          | - Data cleaning and processing  <br/> - Model building                                            | - Machine-learning-based toolkit for the processing of natural language text <br/> - Tokenization, sentence segmentation, part-of-speech tagging, and other tasks <br/> - Built-in tools for data modeling and model validation                                                                                                                                                                             |
+| [Charts](https://github.com/HanSolo/charts)                            | Data exploration and visualization                                                                | - Scientific [JavaFX](https://openjfx.io/) charting library under development                                                                                                                                                                                                                                                                                                                               |
```
Comment (AlejandraPedroza) [inline docs/topics/data-analysis/data-science-libraries.md:41]: Note: The last update for this library was 2 years ago

File: docs/topics/data-analysis/data-science-libraries.md [hunk 6]
```diff
@@ -1,88 +1,43 @@
...
+| [Kotlin for Apache Spark](https://github.com/JetBrains/kotlin-spark-api)     | - Data collection <br/> - Data cleaning and processing <br/> - Data exploration and visualization <br/> - Model building | - Layer of compatibility between [Apache Spark](https://spark.apache.org/) and Kotlin <br/> - Apache Spark data transformation operations in Kotlin-idiomatic code <br/> - Simple usage of Kotlin features, such as data classes and lambda expressions, in curly braces or method reference                                                                                                                                                                                                                                                                                                                                                                     |
```
Comment (zaleslaw) [inline docs/topics/data-analysis/data-science-libraries.md:14]: The order should be following:
Kotlin DataFrame
Kandy
KotlinDL
Multik
Kotlin for Apache Spark
Letsplot

Comment (zaleslaw) [reply docs/topics/data-analysis/data-science-libraries.md:14]: There are our libraries, supported and maintained in JetBrains

File: docs/topics/data-analysis/data-science-libraries.md (hunk 1 above)
Comment (zaleslaw) [reply docs/topics/data-analysis/data-science-libraries.md:17]: Remove please londogard

File: docs/topics/data-analysis/data-science-libraries.md [hunk 7]
```diff
@@ -1,88 +1,43 @@
...
+| [kravis](https://github.com/holgerbrandl/kravis)                             | Data exploration and visualization                                                                                       | - Visualization of tabular data <br/> - Inspired by R's [ggplot](https://ggplot2.tidyverse.org/) <br/> - Support for [Jupyter with Kotlin Kernel](https://github.com/Kotlin/kotlin-jupyter#readme)                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
```
Comment (zaleslaw) [inline docs/topics/data-analysis/data-science-libraries.md:20]: keep kravis at the bottom for a while

File: docs/topics/data-analysis/data-science-libraries.md (hunk 2 above)
Comment (zaleslaw) [reply docs/topics/data-analysis/data-science-libraries.md:31]: Remove dex

File: docs/topics/data-analysis/data-science-libraries.md [hunk 8]
```diff
@@ -1,88 +1,43 @@
...
+| [ND4J](https://github.com/eclipse/deeplearning4j/tree/master/nd4j)     | - Data cleaning and processing  <br/> - Model building                                            | - Matrix mathematics library for the JVM <br/> - Over 500 mathematical, linear algebra, and deep learning operations                                                                                                                                                                                                                                                                                        |
+| [Apache Mahout](https://mahout.apache.org/)                            | - Data cleaning and processing  <br/> - Model building                                            | - Distributed linear algebra framework and mathematically expressive [Scala](https://www.scala-lang.org/api/current/) DSL <br/> - Support for multiple distributed backends (including [Apache Spark](https://spark.apache.org/))  <br/> - Modular native solvers for CPU/GPU/CUDA acceleration                                                                                                             |
```
Comment (zaleslaw) [inline docs/topics/data-analysis/data-science-libraries.md:37]: Remove Apache Mahout

File: docs/topics/data-analysis/data-science-libraries.md (hunk 3 above)
Comment (zaleslaw) [reply docs/topics/data-analysis/data-science-libraries.md:34]: Let's keep

File: docs/topics/data-analysis/data-science-libraries.md (hunk 4 above)
Comment (zaleslaw) [reply docs/topics/data-analysis/data-science-libraries.md:35]: Let's keep

File: docs/topics/data-analysis/data-science-libraries.md (hunk 5 above)
Comment (zaleslaw) [reply docs/topics/data-analysis/data-science-libraries.md:41]: Let's keep

**Review by zaleslaw — CHANGES_REQUESTED** [review]
I am ok with structure and descriptions, but please rearrange the order and remove a few outdated and not-popular items

File: docs/topics/data-analysis/data-science-libraries.md [hunk 9]
```diff
@@ -1,88 +1,40 @@
...
+| [KotlinDL](https://github.com/jetbrains/kotlindl)                            | Model building                                                                                                           | - Deep learning API written in Kotlin and inspired by [Keras](https://keras.io/) <br/> - Training deep learning models from scratch or importing existing Keras and ONNX models for inference <br/> - Transfer learning for tailoring existing pre-trained models to your tasks <br/> - Support for [Jupyter Notebook](https://jupyter.org/) and the [Android platform](https://developer.android.com/about)                                                                                                                                                                                                                                                     |
```
Comment (zaleslaw) [inline docs/topics/data-analysis/data-science-libraries.md:15]: Remove please for [Jupyter Notebook](https://jupyter.org/) (stay only with Android)

File: docs/topics/data-analysis/data-science-libraries.md [hunk 10]
```diff
@@ -1,88 +1,40 @@
...
+| [Kotlin DataFrame](https://github.com/Kotlin/dataframe)                      | - Data collection <br/> - Data cleaning and processing                                                                   | - Operations for creating, sorting, and cleaning data frames, feature engineering, and more  <br/> - Processing of structured data  <br/> - Support for `CSV`, `JSON`, and other input formats <br/> - Reading from SQL databases                                                                                                                                                                                                                                                                                                                                                                                                                               |
```
Comment (daniCsorbaJB) [inline docs/topics/data-analysis/data-science-libraries.md:13]: I think we could add the ability to connect to APIs

File: docs/topics/data-analysis/data-science-libraries.md [hunk 11]
```diff
@@ -1,88 +1,40 @@
...
+| [KotlinDL](https://github.com/jetbrains/kotlindl)                            | Model building                                                                                                           | - Deep learning API written in Kotlin and inspired by [Keras](https://keras.io/) <br/> - Training deep learning models from scratch or importing existing Keras and ONNX models for inference <br/> - Transfer learning for tailoring existing pre-trained models to your tasks <br/> - Support for the [Android platform](https://developer.android.com/about)                                                                                                                                                                                                                                                     |
```
Comment (daniCsorbaJB) [inline docs/topics/data-analysis/data-science-libraries.md:15]: regarding the part: Transfer learning for tailoring existing pre-trained models to your tasks:
Before this cell we used the gerund form of verbs to describe supported features i.e.:  "Reading from.." "Processing structured data". 

I would suggest keeping to one style

File: docs/topics/data-analysis/data-science-libraries.md [hunk 12]
```diff
@@ -1,88 +1,40 @@
...
+| [KMath](https://github.com/mipt-npm/kmath)                                   | - Data cleaning and processing <br/> - Data exploration and visualization <br/> - Model building                         | - Modular library to work with mathematical abstractions in [Kotlin Multiplatform](https://www.jetbrains.com/kotlin-multiplatform/) (JVM, JS, Native, and Wasm) <br/> - APIs for algebraic structures, mathematical expressions, histograms, and streaming operations <br/> - Interchangeable wrappers over existing Java and Kotlin libraries, including [ND4J](https://github.com/eclipse/deeplearning4j/tree/master/nd4j), [Apache Commons Math](https://commons.apache.org/proper/commons-math/), [Multik](https://github.com/Kotlin/multik), and others <br/> - Inspired by Python's [NumPy](https://numpy.org/) but allowing superior architecture designs |
```
Comment (daniCsorbaJB) [inline docs/topics/data-analysis/data-science-libraries.md:19]: Hmm I'm confused about the last point here. What superior architectural design is allowed? Could we name at least one example in a concise way?

File: docs/topics/data-analysis/data-science-libraries.md [hunk 13]
```diff
@@ -1,88 +1,40 @@
...
+| [CoreNLP](https://stanfordnlp.github.io/CoreNLP/)                      | Data cleaning and processing                                                                      | - Natural language processing toolkit <br/> - Linguistic annotations for text, such as sentiment and quote attributions <br/> - Support for eight languages                                                                                                                                                                                                                                                 |
+| [Smile](https://github.com/haifengl/smile)                             | - Data cleaning and processing <br/> - Data exploration and visualization <br/> - Model building  | - Ready-made algorithms for machine learning and natural language processing <br/> - Linear algebra, graph, interpolation, and visualization tools <br/> - Smile provides a functional [Kotlin API](https://github.com/haifengl/smile/tree/master/kotlin), [Scala API](https://github.com/haifengl/smile/tree/master/scala), [Clojure API](https://github.com/haifengl/smile/tree/master/clojure), and more |
```
Comment (daniCsorbaJB) [inline docs/topics/data-analysis/data-science-libraries.md:31]: We never repeat the tool's name in the descriptions for other tools, so I suggest omitting it here as well. (i.e.: Smile provides --> provides)

File: docs/topics/data-analysis/data-science-libraries.md [hunk 14]
```diff
@@ -1,88 +1,40 @@
...
+| [Apache OpenNLP](https://opennlp.apache.org/)                          | - Data cleaning and processing  <br/> - Model building                                            | - Machine-learning-based toolkit for the processing of natural language text <br/> - Tokenization, sentence segmentation, part-of-speech tagging, and other tasks <br/> - Built-in tools for data modeling and model validation                                                                                                                                                                             |
+| [Charts](https://github.com/HanSolo/charts)                            | Data exploration and visualization                                                                | - Scientific [JavaFX](https://openjfx.io/) charting library under development                                                                                                                                                                                                                                                                                                                               |
```
Comment (daniCsorbaJB) [inline docs/topics/data-analysis/data-science-libraries.md:38]: This one sticks out a little compared to the other tools (its only feature being that something is under development)

File: docs/topics/data-analysis/data-science-libraries.md [hunk 15]
```diff
@@ -1,88 +1,40 @@
...
-  JVM and native computational engines, and a combination of the two for optimal performance.
-
-* [KotlinDL](https://github.com/jetbrains/kotlindl) is a high-level Deep Learning API written in Kotlin and inspired
-  by Keras. It offers simple APIs for training deep learning models from scratch, importing existing Keras models
-  for inference, and leveraging transfer learning for tweaking existing pre-trained models to your tasks.
-
-* [Kotlin for Apache Spark](https://github.com/JetBrains/kotlin-spark-api) adds a missing layer of compatibility between
-  Kotlin and Apache Spark. It allows Kotlin developers to use familiar language features such as data classes, and
-  lambda expressions as simple expressions in curly braces or method references.
+In addition, Kotlin has 100% interoperability with Java. This interoperability provides the ability to leverage
```
Comment (daniCsorbaJB) [inline docs/topics/data-analysis/data-science-libraries.md:6]: (Optional: just a thought to further highlight why this interoperability is so amazing - not only does the user get to use loads of Kotlin libraries but they can use Java as well)

In addition to its own libraries, Kotlin has 100% interoperability with Java.

**Review by daniCsorbaJB — COMMENTED** [review]
Looks really good - nice job! 👍 🚀  (I was surprised by the amount of libraries 😮

File: docs/topics/data-analysis/data-science-libraries.md (hunk 12 above)
Comment (AlejandraPedroza) [reply docs/topics/data-analysis/data-science-libraries.md:19]: Is it clearer?

File: docs/topics/data-analysis/data-science-libraries.md (hunk 14 above)
Comment (AlejandraPedroza) [reply docs/topics/data-analysis/data-science-libraries.md:38]: I added more details.

File: docs/topics/data-analysis/data-science-libraries.md (hunk 15 above)
Comment (AlejandraPedroza) [reply docs/topics/data-analysis/data-science-libraries.md:6]: Added it. What do you think?

Comment (daniCsorbaJB) [reply docs/topics/data-analysis/data-science-libraries.md]: lgtm 👍

**Review by daniCsorbaJB — APPROVED** [review]
Thank you for the changes - LGTM 🚀

**Pull request merged**